processor/src/test/java/org/mapstruct/ap/test/source/presencecheck/spi/SoccerTeamMapperNestedObjects.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.source.presencecheck.spi;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Mappings;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Cindy Wang
+ */
+@Mapper
+public interface SoccerTeamMapperNestedObjects {
+    SoccerTeamMapperNestedObjects INSTANCE = Mappers.getMapper( SoccerTeamMapperNestedObjects.class );
+
+    @Mappings({
+        @Mapping(target = "players", ignore = true),
+        @Mapping(target = "goalKeeperName", source = "goalKeeper.name")
```

Comment:
just out of curiosity: what would happen if the target is also nested? Should we add an `@Mapping` for this as well just to be sure it works? So: `@Mapping( target = "referee.name", source = "refereeName" )`